alpine build failure splice

### Issue and Steps to Reproduce
<!-- Describe your issue and tell us how to reproduce it (include any useful information). -->

commit: cb1bd82c86bd09c242523e2c75041a77e7fc9882

while building on alpine docker base image the build fails with following error

```bash
378.6 In file included from common/splice_script.c:15:
378.6 /usr/include/sys/errno.h:1:2: error: #warning redirecting incorrect #include <sys/errno.h> to <errno.h> [-Werror=cpp]
378.6     1 | #warning redirecting incorrect #include <sys/errno.h> to <errno.h>
378.6       |  ^~~~~~~
380.2 cc1: all warnings being treated as errors
380.2 make: *** [Makefile:311: common/splice_script.o] Error 1
```

I don't think this is related to missing dependency but more on the side of using the right header.

for reference the docker image steps are below
```dockerfile
RUN apk update && \
    apk add \
      alpine-sdk \
      autoconf \
      automake \
      ca-certificates \
      cargo \
      gettext \
      git \
      gmp-dev \
      jq \
      libsodium \
      libtool \
      net-tools \
      postgresql-dev \
      py3-mako \
      py3-pip \
      python3 \
      python3-dev \
      sqlite-dev \
      sqlite-static \
      zlib-dev \
      zlib-static \
      linux-headers

RUN pip3 install grpcio-tools

ARG CORE_LIGHTNING_GIT_HASH \
    CORE_LIGHTNING_REPO
RUN git clone --recursive $CORE_LIGHTNING_REPO core-lightning
RUN cd core-lightning && \
    git checkout $CORE_LIGHTNING_GIT_HASH && \
    ./configure --enable-static --prefix=/usr && \
    make -j $(nproc) install && \
    make clean
```

--- a/common/splice_script.c
+++ b/common/splice_script.c
@@ -11,8 +11,8 @@
 #include <common/json_parse_simple.h>
 #include <common/splice_script.h>
 #include <ctype.h>
+#include <errno.h>
 #include <inttypes.h>
-#include <sys/errno.h>
 
 #define SCRIPT_DUMP_TOKENS 0
 #define SCRIPT_DUMP_SEGMENTS 0

